pkg/querier/frontend/frontend.go

```diff
@@ -359,13 +359,17 @@ FindQueue:
 		return nil, err
 	}
 
-	i, n := 0, rand.Intn(len(f.queues))
-	for userID, queue := range f.queues {
-		if i < n {
-			i++
+	keys := make([]string, 0, len(f.queues))
+	for k := range f.queues {
+		keys = append(keys, k)
+	}
+	rand.Shuffle(len(keys), func(i, j int) { keys[i], keys[j] = keys[j], keys[i] })
```

Comment:
Can you add a quick comment explaining why we do a random sort here.

Reply:
This comment already exists at the top of the function:

https://github.com/cortexproject/cortex/blob/d18a3b6b1e4cd3dc99edd6793265380698b3418c/pkg/querier/frontend/frontend.go#L347

This feels redundant, but I can add a comment if you'd like.

Reply:
In that case all good!